03_it_analysis/src/it_analysis_data_prep.py

```diff
@@ -112,39 +111,40 @@ def lag_sources(n_lags, srcs_list):
         srcs_list[s] = srcs_list[s].sort_index(axis=1)
     return srcs_list
 
-#%%
-def create_correlation_matrix(srcs_list_lagged, snks_list):
-    '''Takes in the lagged sources and calculates the correlation between them and the
-    sinks. Pearson correlation is used'''
-    corrs = pd.DataFrame()
-    for sc in range(len(srcs_list_lagged)):
-        col_snks = pd.DataFrame()
-        for sk in range(len(snks_list)):
-            sc_sk_corr = srcs_list_lagged[sc].apply(lambda s: snks_list[sk].corrwith(s))
-            #print(sc_sk_corr)
-            #corrs.append(sc_sk_corr)
-            col_snks = col_snks.append(sc_sk_corr)
-        corrs[list(col_snks.columns)] = col_snks
-    return corrs
-
-#%%
-def plot_heat_map(corrs, mask_threshold, date_start, date_end, save_location, save=False):
-    '''Takes in a correlation matrix and plots a heat map of correlations,
-    mask_threshold is a threshold where if abs(correlation) < threshold those
-    squares of the heatmap are masked out to highlight stronger correlation. save is
-    true/false whether it will save the plot to file, default false'''
+def main():
+    # import config
+    with open("03_it_analysis/it_analysis_data_prep_config.yaml", 'r') as stream:
+        config = yaml.safe_load(stream)['it_analysis_data_prep.py']
+        
+    #select the sources
+    srcs = config['srcs']
+    #select the sinks
+    snks = config['snks']
+    #start date for analysis
+    date_start = config['date_start']
+    #end date for analysis
+    date_end = config['date_end']
+    #number of lag days to consider for sources
+    n_lags = config['n_lags']
+    #generate the sources data
+    srcs_list = select_sources(srcs, date_start, date_end)    
+    #generate the sinks data
+    snks_list = select_sinks(snks, date_start, date_end)
+    #lag the sources
+    srcs_list_lagged = lag_sources(n_lags, srcs_list)
+    #output file
+    out_dir = config['out_dir']
+    #create out directory if it doesn't already exist
+    os.makedirs(out_dir, exist_ok = True)
+    #write snks_list to file 
+    snks_file = open(out_dir+'snks', "wb")
+    pickle.dump(snks_list, snks_file)
+    snks_file.close()
+    #write srcs_list_lagged to file 
+    srcs_file = open(out_dir+'srcs_lagged', "wb")
+    pickle.dump(srcs_list_lagged, srcs_file)
+    srcs_file.close()
```

Comment:
Could these be included in the select_sinks() and lag_sources() functions? Or perhaps create a new function that takes in a list of lists and file names and saves the output? Just trying to think about how to keep the main() function as concise as possible

Reply:
Yeah, this is a good idea, I'll move the save calls from `main()` into their specific functions. Good idea.